python/perf-kernels/streamk/streamk_kernel.py

```diff
@@ -166,25 +146,34 @@ def streamk_gemm(
             B_BASE += BLOCK_SIZE_K * stride_bk
 
         tile_iter = tile_id * iters_per_tile
-        if start_iter == tile_iter:
-            tile_iter_end = tile_iter + iters_per_tile
+
+        if start_iter != tile_iter:
+            rm1 = tl.arange(0, BLOCK_SIZE_M)
+            rn1 = tl.arange(0, BLOCK_SIZE_N)
+            rm1 = tl.max_contiguous(tl.multiple_of(rm1, BLOCK_SIZE_M), BLOCK_SIZE_M)
+            rn1 = tl.max_contiguous(tl.multiple_of(rn1, BLOCK_SIZE_N), BLOCK_SIZE_N)
+            P_ = P + pid * BLOCK_SIZE_M * BLOCK_SIZE_N + rm1[:, None] * BLOCK_SIZE_N + rn1[None, :]
+            tl.store(P_, acc, cache_modifier=".wt")
+            tl.store(locks + pid, 1, cache_modifier=".wt")
+        else:
             next_pid = pid + 1
+            tile_iter_end = tile_iter + iters_per_tile
             end = end_iter
-            while (end < tile_iter_end and next_pid < num_cus):
-                # todo: try use tl.load once cache modifier landed upstream
-                while tl.atomic_cas(locks + next_pid, 1, 1) != 1:
+            while (end < tile_iter_end and next_pid < NUM_SMS):
+                while tl.load(locks + next_pid, cache_modifier=".cv", volatile=True) != 1:
```

Comment:
This also does not work in `gfx90a`: https://github.com/ROCm/triton-internal/issues/311